worker/peergrouper/worker_test.go

```diff
@@ -687,13 +687,36 @@ func assertMemberAddresses(c *gc.C, st *fakeState, addrTemplate string, addrDesi
 	c.Check(obtained, gc.DeepEquals, expected)
 }
 
-func (s *workerSuite) TestReturnsErrorWhenNoHASpaceAndMachinesWithMultiLocalAddr(c *gc.C) {
-	DoTestForIPv4AndIPv6(c, s, func(ipVersion TestIPVersion) {
-		st := haSpaceTestCommonSetup(c, ipVersion)
-		err := s.newWorker(c, st, st.session, nopAPIHostPortsSetter{}).Wait()
-		errMsg := `computing desired peer group: machine "1[012]" has more than one non-local address and juju-ha-space is not set`
-		c.Check(err, gc.ErrorMatches, errMsg)
-	})
+func (s *workerSuite) TestReturnsErrorForNewPeersAndNoHASpaceAndMachinesWithMultiAddrIPv4(c *gc.C) {
+	s.doTestReturnsErrorForNewPeersAndNoHASpaceAndMachinesWithMultiAddr(c, testIPv4)
+}
+
+func (s *workerSuite) TestReturnsErrorForNewPeersAndNoHASpaceAndMachinesWithMultiAddrIPv6(c *gc.C) {
+	s.doTestReturnsErrorForNewPeersAndNoHASpaceAndMachinesWithMultiAddr(c, testIPv6)
+}
+
+func (s *workerSuite) doTestReturnsErrorForNewPeersAndNoHASpaceAndMachinesWithMultiAddr(
+	c *gc.C, ipVersion TestIPVersion,
+) {
+	st := haSpaceTestCommonSetup(c, ipVersion, "0v")
+	err := s.newWorker(c, st, st.session, nopAPIHostPortsSetter{}).Wait()
+	errMsg := `computing desired peer group: updating member address: ` +
+		`machine "1[12]" has more than one non-local address and juju-ha-space is not set`
```

Comment:
presumably this is saying "machine 11" or "machine 12" ?
I was a bit confused at first.

Reply:
Yep; regex to mitigate potential intermittent failures.